lib/modules/manager/gradle/parser.ts

```diff
@@ -1,1208 +1,458 @@
-      {
-        matchType: [TokenType.String, TokenType.StringInterpolation],
-        tokenMapKey: 'scriptFile',
-      },
-      { matchType: TokenType.RightParen },
-    ],
-    handler: processApplyFrom,
-  },
-];
-
-function tryMatch({
-  tokens,
-  variables,
-  packageFile,
-}: MatchConfig): SyntaxHandlerOutput {
-  for (const { matchers, handler } of matcherConfigs) {
-    const tokenMap = matchTokens(tokens, matchers);
-    if (tokenMap) {
-      const result = handler({
-        packageFile,
-        variables,
-        tokenMap,
-      });
-      if (result !== null) {
-        return result;
-      }
-    }
-  }
-  tokens.shift();
-  return null;
-}
-
-async function parseInlineScriptFile(
-  scriptFile: string,
-  variables: PackageVariables,
-  recursionDepth: number,
-  packageFile = ''
-): Promise<SyntaxHandlerOutput> {
-  if (recursionDepth > 2) {
-    logger.debug(`Max recursion depth reached in script file: ${scriptFile}`);
-    return null;
-  }
-
-  if (!regEx(/\.gradle(\.kts)?$/).test(scriptFile)) {
-    logger.warn({ scriptFile }, `Only Gradle files can be included`);
-    return null;
-  }
-
-  const scriptFilePath = getSiblingFileName(packageFile, scriptFile);
-  const scriptFileContent = await readLocalFile(scriptFilePath, 'utf8');
-  if (!scriptFileContent) {
-    logger.debug(`Failed to process included Gradle file ${scriptFilePath}`);
-    return null;
-  }
-
-  return parseGradle(
-    scriptFileContent,
-    variables,
-    scriptFilePath,
-    recursionDepth + 1
-  );
-}
-
-export async function parseGradle(
+  storeVarToken(ctx, nodeTransformed);
+  return ctx;
+});
+
+// foo.bar["baz"] = "1.2.3"
+const qVariableAssignmentIdentifier = q
+  .sym(storeVarToken)
+  .many(
+    q.alt(
+      q.op<Ctx>('.').sym(storeVarToken),
+      q.tree<Ctx>({
+        type: 'wrapped-tree',
+        maxDepth: 1,
+        startsWith: '[',
+        endsWith: ']',
+        search: q.begin<Ctx>().join(qStringValueAsSymbol).end(),
+      })
+    ),
+    0,
+    32
+  )
+  .handler(stripReservedPrefixFromKeyTokens);
+
+// foo.bar["baz"] -> "foo.bar.baz"
+const qVariableAccessIdentifier =
+  qVariableAssignmentIdentifier.handler(coalesceVariable);
+
+// "foo${bar}baz"
+const qTemplateString = q
+  .tree({
+    type: 'string-tree',
+    maxDepth: 2,
+    preHandler: (ctx) => {
+      ctx.tmpTokenStore.templateTokens = [];
+      return ctx;
+    },
+    search: q.alt(
+      qStringValue.handler((ctx) => {
+        ctx.tmpTokenStore.templateTokens?.push(...ctx.varTokens);
+        ctx.varTokens = [];
+        return ctx;
+      }),
+      qVariableAccessIdentifier.handler((ctx) => {
+        ctx.tmpTokenStore.templateTokens?.push(...ctx.varTokens);
+        ctx.varTokens = [];
+        return ctx;
+      })
+    ),
+  })
+  .handler((ctx) => {
+    ctx.varTokens = ctx.tmpTokenStore.templateTokens!;
+    return ctx;
+  });
+
+// foo = "1.2.3"
+const qGroovySingleVarAssignment = qVariableAssignmentIdentifier
```

Comment:
should we split groovy and kotlin langs for better maintainabillity and no false positives?

We should at least split this file to common / groovy and kotlin parts. 🤔  Can also be done in a follow up pr.

Reply:
My preference would have been to group these blocks "feature-wise", e.g., everything assignment-related into one file, everything version-catalog related into another, ... because sometimes Groovy / Kotlin parts differ just by a delimiter like `:` and `=`

I didn't do so because as soon as I try `export` on queries, I get: 
`TS4094: Property 'builders' of exported class expressions may not be private or protected.`
To my understanding, `builders` would need to be [made public](https://github.com/zharinov/good-enough-parser/blob/main/lib/query/builder.ts#L230) in good-enough-parser but presumably there was a good reason to not expose it.

Reply:
@zharinov ideas? as this file is pretty big.

Reply:
But I have no idea what `TS4096` actually means

Reply:
The reason of not exposing `builders` was to have only public method that compose query, e.g. `sym('foo').op('+').sym('bar')` expands to `SeqBuilder { builders: [SymBuilder, OpBuilder, SymBuilder] }`

Reply:
Okay, I think I found the solution, let me check it

Reply:
✅ Fixed in the `1.1.21` version